ruapu.h

```diff
@@ -183,8 +183,13 @@ RUAPU_INSTCODE(avx512ifma, 0x62, 0xf2, 0xfd, 0x48, 0xb4, 0xc0) // vpmadd52luq zm
 RUAPU_INSTCODE(avx512vbmi, 0x62, 0xf2, 0x7d, 0x48, 0x75, 0xc0) // vpermi2b zmm0,zmm0,zmm0
 RUAPU_INSTCODE(avx512vbmi2, 0x62, 0xf2, 0x7d, 0x48, 0x71, 0xc0) // vpshldvd zmm0,zmm0,zmm0
 RUAPU_INSTCODE(avx512fp16, 0x62, 0xf6, 0x7d, 0x48, 0x98, 0xc0) // vfmadd132ph zmm0,zmm0,zmm0
+RUAPU_INSTCODE(avx512pf, 0x62, 0xf2, 0xfd, 0x48, 0xc8, 0xc0) //vgatherpf1dps DWORD PTR [rsi+zmm0*1]{k1}
+RUAPU_INSTCODE(avx512er, 0xc7, 0x45, 0xfc, 0x00, 0x00, 0x00, 0x00) //vexp2pd zmm0,zmm0
+RUAPU_INSTCODE(avx5124fmaps, 0x62, 0xf2, 0x7f, 0x48, 0x52, 0x08) //v4fmaddps zmm1,zmm0,XMMWORD PTR [rax]
+RUAPU_INSTCODE(avx5124vnniw, 0xc7, 0x45, 0xfc, 0x00, 0x00, 0x00, 0x00) //vp4dpwssd zmm1,zmm0,XMMWORD PTR [rax]
```

Comment:
do not touch memory for arithmetic instructions
for memory load, use stack pointer, sp/esp/rsp